cmd/apiserver/server.go

```diff
@@ -60,12 +56,84 @@ func (s *server) createCalculationBulk(c *gin.Context) {
 	}
 
 	if err := s.client.Create(s.ctx, bulk); err != nil {
-		responseError(c, "couldn't create calculation", err)
+		responseError(c, "couldn't create calculation bulk", err)
 	} else {
 		c.JSON(http.StatusOK, gin.H{"data": bulk})
 	}
 }
 
+func (s *server) createWorkerPool(c *gin.Context) {
+	body, err := ioutil.ReadAll(c.Request.Body)
+	if err != nil {
+		responseError(c, "couldn't read body", err)
+		return
+	}
+
+	workerPoolName := fmt.Sprintf("workerpool-%s", util.InputHash(body))
+	var workerPools struct {
+		Workers        map[string]workersv1.Worker `json:"workers,omitempty"`
+		CompletionTime metav1.Time                 `json:"completionTime,omitempty"`
+		CreationTime   metav1.Time                 `json:"creationTime,omitempty"`
+		PendingTime    metav1.Time                 `json:"pendingTime,omitempty"`
+	}
+
+	if err := json.Unmarshal(body, &workerPools); err != nil {
+		responseError(c, "couldn't unmarshal body", err)
+	}
+
+	s.logger.Info("Creating the workerpool...")
+
+	wp := &workersv1.WorkerPool{
+		ObjectMeta: metav1.ObjectMeta{Name: workerPoolName, Namespace: s.namespace},
+		Spec: workersv1.WorkerPoolSpec{
+			Workers: workerPools.Workers,
+		},
+		Status: workersv1.WorkerPoolStatus{
+			CreationTime:   &workerPools.CreationTime,
+			PendingTime:    &workerPools.PendingTime,
+			CompletionTime: &workerPools.CompletionTime,
+		},
+	}
+
+	if err := s.client.Create(s.ctx, wp); err != nil {
+		responseError(c, "couldn't create calculation bulk", err)
+	} else {
+		c.JSON(http.StatusOK, gin.H{"data": wp})
+	}
+}
+
+func (s *server) deleteCalculationBulk(c *gin.Context) {
+	calcBulkName := c.Param("id")
+	bulk := &bulkv1.CalculationBulk{}
+	err := s.client.Get(s.ctx, ctrlruntimeclient.ObjectKey{Namespace: s.namespace, Name: calcBulkName}, bulk)
```

Comment:
We should retry on conflict here. Check example in: https://github.com/vega-project/ccb-operator/blob/master/pkg/util/workers.go#L65